internal/session/sessionwants.go

```diff
@@ -11,19 +11,27 @@ import (
 // sessionWants keeps track of which cids are waiting to be sent out, and which
 // peers are "live" - ie, we've sent a request but haven't received a block yet
 type sessionWants struct {
-	toFetch   *cidQueue
-	liveWants map[cid.Cid]time.Time
+	// The wants that have not yet been sent out
+	toFetch *cidQueue
+	// Wants that have been sent but have not received a response
+	liveWants *cidQueue
```

Comment:
This is going to perform poorly. `cidQueue` works well when we add/remove using `Pop` and `Push`, but not when removing with `Remove`. We're never going to remove anything unless we cal `Pop` or `Cids` (broadcast).